lightly/api/utils.py

```diff
@@ -45,17 +45,20 @@ def retry(func, *args, **kwargs):
     max_retries = RETRY_MAX_RETRIES
 
     # try to make the request
-    for i in range(max_retries):
+    current_retries = 0
+    while True:
         try:
             # return on success
             return func(*args, **kwargs)
-        except Exception:
+        except Exception as e:
             # sleep on failure
             time.sleep(backoff)
             backoff = 2 * backoff if backoff < max_backoff else backoff
-        
-    # max retries exceeded
-    raise RuntimeError('The connection to the server timed out.')
+            current_retries += 1
+
+            # max retries exceeded
+            if current_retries >= max_retries:
+                raise RuntimeError from e
```

Comment:
I would keep some message in the `RuntimeError` e.g.
```python
raise RuntimeError('Maximum retries exceeded!') from e
```

Reply:
I changed it to
`raise RuntimeError(f'Maximum retries exceeded! Original exception: {type(e)}: {str(e)}') from e`

Reply:
No, otherwise the original exception is not printed if you do `str(exception)`, like we do in the docker. It is only printed in the full print modes that also print the `exception.__cause__`.